### PR TITLE
Port to dune

### PR DESCRIPTION
**Pull request by pascutto**

### CODE REVIEW

**Review by emillon — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 _build/
+*.merlin
+*.install
 setup.bin
 setup.data
 setup.log
```
Comment (hannesm) [inline .gitignore]: the `setup*` entries can be removed now (leftover from early `_oasis` days)

File: dune-project [hunk 2]
```diff
@@ -0,0 +1,3 @@
+(lang dune 1.7)
+
+(name mirage-block-solo5)
```
Comment (hannesm) [inline dune-project:2]: there's an unneeded newline in this file.
would `dune 1.0` suffice here, or are there any 1.7 features strongly necessary?

File: mirage-block-solo5.opam [hunk 3]
```diff
@@ -7,14 +7,16 @@ authors:      ["Dan Williams" "Martin Lucina"]
...
+ ["dune" "build" "-p" name "-j" jobs]
+ ["dune" "runtest" "-p" name] {with-test}
 ]
+
 depends: [
   "ocaml" {>= "4.04.2"}
-  "ocamlfind"  {build}
-  "ocamlbuild" {build}
-  "topkg"      {build}
+  "dune"  {build & >= "1.7.0"}
```
Comment (hannesm) [inline mirage-block-solo5.opam]: I'd be in favour, as above, to use `dune 1.0.0`

**Review by hannesm — CHANGES_REQUESTED** [review]
thanks, I've some minor nits. I'm also not sure whether `mirage-solo5` is needed in the `src/dune` file (it is a dependency to get the used C symbols, which are defined by `mirage-solo5`)

File: dune-project (hunk 2 above)
Comment (pascutto) [reply dune-project:2]: `dune 1.0` is probably enough :)

**Review by hannesm — APPROVED** [review]

**Comment by mato** [comment]
Thank you for your PR. I have two comments:

1. It is a bit impolite to change the build system used for a package,
without so much as pinging the person listed as "maintainer" first.

2. Given that you have changed the build system, can you give me a
step-by-step workflow for publishing a new release of this package with
dune, i.e. the equivalent of the steps I would have been using with "topkg"
until now?

**Comment by hannesm** [comment]
hi @mato. for (1) I agree, but in the MirageOS organization we've a pretty bad track record of maintaining the author and maintainer metadata of opam packages (yes, your mirage-*-solo5 packages are an exception to that). we should also fix the metadata (maybe using [code owners](https://help.github.com/en/articles/about-code-owners), but certainly a way to specify them only once (and not in codeowners and in opam) would be beneficial).
for (2): similar to the opam package `topkg-care` there is the `dune-release` package in opam, which has the same workflow:
- dune-release tag
- dune-release distrib
- dune-release publish
- dune-release opam pkg
- dune-release opam submit

**Comment by pascutto** [comment]
I'm really sorry, I was too quick on that, won't happen again !

**Comment by mato** [comment]
(Sorry for the brevity, replying on mobile)

Clement: ‎My previous reply was ambiguous. This is not your fault, don't worry about it.

Hannes: Ok, thanks. I'll try that workflow.

**Pull request merged**